### PR TITLE
Copter: slow auto descent sets throttle mix to minimum

### PR DESCRIPTION
**Pull request by jschall**

### CODE REVIEW

File: ArduCopter/land_detector.cpp [hunk 1]
```diff
@@ -174,10 +190,16 @@ void Copter::update_throttle_thr_mix()
...
         bool descent_not_demanded = pos_control.get_desired_velocity().z >= 0.0f;
 
-        if ( large_angle_request || large_angle_error || accel_moving || descent_not_demanded) {
-            attitude_control.set_throttle_mix_max();
-        } else {
+        bool xy_speed_low = (position_ok() || optflow_position_ok()) && xy_speed_cms <= 125.0f;
+        bool xy_speed_demand_low = pos_control.is_active_xy() && xy_des_speed_cms <= 125.0f;
+        bool slow_descent_demanded = pos_control.is_active_z() && des_climb_rate_cms < 0.0f && des_climb_rate_cms >= -100.0f;
+
+        bool slow_auto_descent = xy_speed_demand_low && xy_speed_low && slow_descent_demanded;
```
Comment (derekxm) [inline ArduCopter/land_detector.cpp]: slow_auto_descent will be true when vehicle is holding altitude in place, for example, waiting for landing approval at a station. What is the affect of false positive and false negative in setting throttle mix?

File: ArduCopter/land_detector.cpp [hunk 2]
```diff
@@ -174,10 +190,16 @@ void Copter::update_throttle_thr_mix()
...
+
+        if ( slow_auto_descent || !(large_angle_request || large_angle_error || accel_moving || descent_not_demanded) ) {
```
Comment (presmini) [inline ArduCopter/land_detector.cpp]: Just making sure we have the logic correct here, slow_auto_descent will be overriden by any of large_angle_request || large_angle_error || accel_moving || descent_not_demanded right? That's what we want?

Comment (jschall) [inline ArduCopter/land_detector.cpp]: The original logic was:
```
if ( large_angle_request || large_angle_error || accel_moving || descent_not_demanded) {
    attitude_control.set_throttle_mix_max();
} else {
    attitude_control.set_throttle_mix_min();
}
```

Let original_criteria = large_angle_request || large_angle_error || accel_moving || descent_not_demanded

```
if (original_criteria) {
    attitude_control.set_throttle_mix_max();
} else {
    attitude_control.set_throttle_mix_min();
}
```

Invert if statement:
```
if (!original_criteria) {
    attitude_control.set_throttle_mix_min();
} else {
    attitude_control.set_throttle_mix_max();
}
```

Add new criteria:
```
if (slow_auto_descent || !original_criteria) {
    attitude_control.set_throttle_mix_min();
} else {
    attitude_control.set_throttle_mix_max();
}
```

Sub out original_criteria:
```
if (slow_auto_descent || !(large_angle_request || large_angle_error || accel_moving || descent_not_demanded)) {
    attitude_control.set_throttle_mix_min();
} else {
    attitude_control.set_throttle_mix_max();
}
```

File: ArduCopter/land_detector.cpp (hunk 1 above)
Comment (jschall) [inline ArduCopter/land_detector.cpp]: It shouldn't be true and it won't matter anyway for hovering/slow flight.

Comment (jschall) [inline ArduCopter/land_detector.cpp]: Addendum: it shouldn't be true because there won't be a demanded descent.

**Pull request merged**